File: backend/src/main/java/com/devpilot/user/application/MeResponse.java
package com.devpilot.user.application;

import com.devpilot.common.security.UserRole;
import com.devpilot.integration.ai.api.AiStatus;
import com.devpilot.user.domain.ExperienceProfile;
import com.devpilot.user.domain.UserStatus;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;
import org.jspecify.annotations.Nullable;

/**
 * {@code GET /me}·{@code PATCH /me} 응답이자 {@code POST /onboarding} 응답의 {@code user} (docs/05 §3.1,
 * §4.1). onboarding 모듈이 같은 타입을 쓰므로 presentation이 아니라 application에 둔다(docs/03 §2.2 규칙 1).
 */
public record MeResponse(
        UUID id,
        String displayName,
        UserRole role,
        UserStatus status,
        String timezone,
        int dayStartHour,
        int weekdayStudyMinutes,
        int weekendStudyMinutes,
        @Nullable ExperienceProfile experienceProfile,
        @Nullable LocalDate experienceStartDate,
        boolean onboardingCompleted,
        @Nullable Instant onboardingCompletedAt,
        LocalDate today,
        boolean calendarSubscribed,
        @Nullable Instant deletionRequestedAt,
        AiStatus aiStatus,
        AiUsageView aiUsage,
        Instant createdAt,
        long version) {}

File: app/lib/features/settings/data/me_response.dart
import 'package:devpilot_app/core/api/api_enums.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'me_response.freezed.dart';
part 'me_response.g.dart';

/// `MeResponse` of `GET/PATCH /me` (docs/05 §3.1). Dates are plan dates (`yyyy-MM-dd`).
@freezed
abstract class MeResponse with _$MeResponse {
  const factory MeResponse({
    required String id,
    required String displayName,
    @JsonKey(unknownEnumValue: UserRole.unknown) required UserRole role,
    @JsonKey(unknownEnumValue: UserStatus.unknown) required UserStatus status,
    required String timezone,
    required int dayStartHour,
    required int weekdayStudyMinutes,
    required int weekendStudyMinutes,
    required bool onboardingCompleted,
    DateTime? onboardingCompletedAt,

    /// Server plan-day of the user; the only "today" the client uses (docs/05 §1.1).
    required String today,
    required bool calendarSubscribed,
    DateTime? deletionRequestedAt,
    @JsonKey(unknownEnumValue: AiStatus.unknown) required AiStatus aiStatus,
    required AiUsageView aiUsage,
    required DateTime createdAt,
    required int version,
  }) = _MeResponse;

  factory MeResponse.fromJson(Map<String, Object?> json) => _$MeResponseFromJson(json);
}

/// `AiUsageView` (docs/05 §3.1). Money is a 2-decimal USD string.
@freezed
abstract class AiUsageView with _$AiUsageView {
  const factory AiUsageView({
    required int todayCalls,
    required int dailyCallLimit,
    required String monthCostUsd,
    required String monthlyBudgetUsd,
  }) = _AiUsageView;

  factory AiUsageView.fromJson(Map<String, Object?> json) => _$AiUsageViewFromJson(json);
}
